app/models/plan.rb

```diff
@@ -189,8 +189,36 @@ def get_guidance_group_options
     end
     return ggroups.uniq
   end
+  
+  ##
+  # Sets up the plan for feedback:
+  #  emails confirmation messages to owners
+  #  emails org admins and org contact 
+  #  adds org admins to plan with the 'reviewer' Role
+  def request_feedback(user)
+    self.feedback_requested = true
+   
+    # Share the plan with each org admin as the reviewer role
+    admins = user.org.org_admins
+    admins.each do |admin|
+      role = Role.create!(plan: self, user: admin, access: 16)
+    end 
+    
+    # Send an email confirmation to the owners and co-owners
+    self.owner_and_coowners.each do |owner|
+      UserMailer.feedback_confirmation(owner, self, user).deliver_now
+    end
+  
+    # Send an email to all of the org admins as well as the Org's administrator email
+    if user.org.contact_email.present?
+      admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+    end
+    admins.each do |admin|
+      UserMailer.feedback_notification(admin, self, user).deliver_now
+    end
 
-
+    self.save!
```

Comment:
this line would be more adequate right after L199, so that if fails saving roles would be no inserted and no one will receive email...

Reply:
move the feedback_requested flag down to the end instead. The idea is that the flag would not be set in the event that any of the other more fragile steps fail (emails and role assignments)